examples/settings.rs

```diff
@@ -0,0 +1,48 @@
+use meilisearch_sdk::client::Client;
+use meilisearch_sdk::indexes::Index;
+use meilisearch_sdk::settings::Settings;
+
+// we need an async runtime
+#[tokio::main(flavor = "current_thread")]
+async fn main() {
+    let client = Client::new("http://localhost:7700", "masterKey");
+
+    // We try to create an index called `movies` with a primary_key of `movie_id`.
+    let my_index: Index = client
+        .create_index("movies", Some("movie_id"))
+        .await
+        .unwrap()
+        // The creation of indexes is asynchronous. But for the sake of the example so we will
+        // wait until the update is entirely processed.
+        .wait_for_completion(&client, None, None)
+        .await
+        .unwrap()
+        // If the creation was successful we can generate an `Index` out of it.
+        .try_make_index(&client)
+        .unwrap();
```

Comment:
I was reading some ["scientific articles"](https://stackoverflow.com/questions/36362020/what-is-unwrap-in-rust-and-what-is-it-used-for)(sarcasm) about `unwrap`. 

My question is, can we use `?` here? And can we avoid these calls or I just don't need to bother with that?

Reply:
Basically, we can use `?` when our function returns a `Result`.

A little rust lesson just so you know :grin: : 
```rust
let result = task.try_make_index(&client);
let index = match result {
    Ok(index) => index,
    Err(error) => return Err(error.into()), // `.into` try to convert the error to the type the current function expect
};
```

This code is equivalent to:
```rust
let index = task.try_make_index(&client)?;
```

-------------

Ok so now you know :grin: 

> My question is, can we use ? here? And can we avoid these calls or I just don't need to bother with that?

We could make our `main` return a `Result<(), meilisearch_sdk::error::Error>` (we can't return anything else than `()` in a `main`).
But in an example _I THINK_, we prefer to write the dumbest code possible to show every little step without any ambiguity.
Now that's my opinion and I can be wrong. There is no standard in rust about that.

Another thing we could do is call `expect` instead of `unwrap`. Expect allows us to write a little error message that could be more explicit for the user before unwrapping the error. Something like `Can't reach the server.`.
Like [here](https://github.com/serenity-rs/serenity/blob/current/examples/e03_struct_utilities/src/main.rs#L36-L46) for example.

Reply:
Thanks a lot for the detailed explanation you're the best! Now it makes sense to me (no, I'm not an expert yet but it makes sense to me haha).
I agree with you regarding the samples, being the simplest as possible. Experienced users will not how to make it simpler :)